### PR TITLE
Move online volume expansion to Beta

### PR DESCRIPTION
**Pull request by gnufied**
xref - kubernetes/enhancements#531

cc @msau42 
/sig storage

replaces - https://github.com/kubernetes/kubernetes/pull/67608

```release-note
Move online volume expansion to beta
```

### CODE REVIEW

**Comment by msau42** [comment]
We should implement the migration logic along with this too so that migration does not fall behind.

**Comment by gnufied** [comment]
@msau42 @leakingtapan do you want me to fix migration logic in this PR for both online and offline resizing? I can keep them as separate commits but keep them in same PR.

**Comment by msau42** [comment]
I would like migration logic to be merged before moving online migration to beta. Whether it's in the same or separate PR doesn't matter to me.

File: test/e2e/storage/volume_expand.go [hunk 1]
```diff
@@ -165,6 +166,48 @@ var _ = utils.SIGDescribe("Volume expand", func() {
...
+		ginkgo.By("Waiting for cloudprovider resize to finish")
+		err = waitForControllerVolumeResize(pvc, c, totalResizeWaitPeriod)
+		framework.ExpectNoError(err, "While waiting for pvc resize to finish")
+
+		ginkgo.By("Waiting for file system resize to finish")
+		pvc, err = waitForFSResize(pvc, c)
+		framework.ExpectNoError(err, "while waiting for fs resize to finish")
+
+		pvcConditions := pvc.Status.Conditions
+		gomega.Expect(len(pvcConditions)).To(gomega.Equal(0), "pvc should not have conditions")
```
Comment (jsafrane) [inline test/e2e/storage/volume_expand.go]: Can you please also do `df` in the pod to check the size is actually bigger?

**Review by jsafrane — COMMENTED** [review]
Looks good enough, IMO it's ready for merge even without the test fix.

**Comment by jsafrane** [comment]
/approve
volume manager changes

**Comment by gnufied** [comment]
@msau42 for completeness sake, PR for handling volume resizing migration is - https://github.com/kubernetes/kubernetes/pull/77994 

@jsafrane I have not *yet* addressed - https://github.com/kubernetes/enhancements/pull/737#discussion_r277895607 in this PR. I think it was agreed that, we can change the implementation of resize check even after implementation merges.

**Comment by jsafrane** [comment]
I checked with @gnufied reworking of DSWP and it's a bit bigger PR. I'd like to defer it to 1.16 as implementation detail, the resize feature won't change its API nor behavior.

/lgtm

**Comment by gnufied** [comment]
@msau42 @jsafrane yeah I have a WIP branch for moving "resizeRequired" error to DSW - https://github.com/gnufied/kubernetes/commit/d9c08d53c0c751731016addaeea79a4c3709bc31 but I need to clean it up and fix and add more tests. I will do that as a follow up.

**Comment by liggitt** [comment]
/test pull-kubernetes-typecheck 

this no longer compiles on master

**Comment by liggitt** [comment]
/hold
to keep fejta-bot from retesting until the compile error is fixed. feel free to remove once it is fixed.

**Comment by gnufied** [comment]
Fixed compilation error.

/hold cancel

**Pull request merged**